src/Functions/ReplaceStringImpl.h

```diff
@@ -346,7 +353,21 @@ struct ReplaceStringImpl
         size_t input_rows_count)
     {
         if (needle.empty())
-            throw Exception(ErrorCodes::ARGUMENT_OUT_OF_BOUND, "Length of the pattern argument in function {} must be greater than 0.", name);
+        {
+            assert(input_rows_count == haystack_data.size() / n);
```

Comment:
`chassert()` is better because it helps us to find problem when the code is run in our CI builds.
`assert()` is no-op in any release builds, whereas `chassert()` still works in release builds with sanitizers which we run in our CI.